framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -2468,77 +2486,90 @@ public AnnotatedTypeMirror getDefaultValueAnnotatedType(TypeMirror typeMirror) {
+   * redundant postconditions (subclasses are permitted to remove these restrictions by overriding
+   * {@link #createRequiresOrEnsuresQualifier}).
    *
-   * <p>This is of the form {@code @EnsuresQualifier(expression="this.elt", qualifier=MyQual.class)}
-   * when elt is declared as {@code @A} or {@code @Poly*} and f contains {@code @B} which is a
-   * sub-qualifier of {@code @A}.
+   * <p>Each annotation in the list is of the form {@code @EnsuresQualifier(expression="expression",
+   * qualifier=MyQual.class)}. {@code expression} must be a valid Java Expression string, in the
+   * same format used by {@link EnsuresQualifier}.
    *
-   * @param elt element for a field
-   * @param inferredType the type of the field, on method exit
+   * @param expression an expression
+   * @param inferredType the type of the expression, on method exit
+   * @param declaredType the declared type of the expression
    * @param preconds the precondition annotations for the method; used to suppress redundant
    *     postconditions
    * @return postcondition annotations for the element (possibly an empty list)
    */
   public final List<AnnotationMirror> getPostconditionAnnotations(
-      VariableElement elt, AnnotatedTypeMirror inferredType, List<AnnotationMirror> preconds) {
-    return getPreOrPostconditionAnnotations(elt, inferredType, BeforeOrAfter.AFTER, preconds);
+      String expression,
+      AnnotatedTypeMirror inferredType,
+      AnnotatedTypeMirror declaredType,
+      List<AnnotationMirror> preconds) {
+    return getPreOrPostconditionAnnotations(
+        expression, inferredType, declaredType, BeforeOrAfter.AFTER, preconds);
   }
 
   /**
-   * Helper method for {@link #getPreconditionAnnotations} and {@link #getPostconditionAnnotations}.
+   * Creates pre- and postcondtion annotations. Helper method for {@link
+   * #getPreconditionAnnotations} and {@link #getPostconditionAnnotations}.
    *
    * <p>Returns a {@code @RequiresQualifier} or {@code @EnsuresQualifier} annotation for the given
-   * field. Returns an empty list if none can be created, because the qualifier has
+   * expression. Returns an empty list if none can be created, because the qualifier has
    * elements/arguments, which {@code @RequiresQualifier} and {@code @EnsuresQualifier} do not
    * support.
    *
    * <p>This implementation makes no assumptions about preconditions suppressing postconditions, but
    * subclasses may do so.
    *
-   * @param elt element for a field
-   * @param inferredType the type of the field, on method entry or exit (depending on the value of
-   *     {@code preOrPost})
+   * @param expression an expression whose type annotations to return
+   * @param inferredType the type of the expression, on method entry or exit (depending on the value
+   *     of {@code preOrPost})
+   * @param declaredType the declared type of the expression, which is used to determine whether the
+   *     inferredType is redundant
```

Comment:
I think you mean that the inferred type supplies no additional information beyond the declared type.  It might be better to say that.  I think "redundant" is not the right word; maybe "vacuous"?  Or just use the definition rather than a word.